sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/GenerateProjection.scala

```diff
@@ -184,7 +187,8 @@ object GenerateProjection extends CodeGenerator[Seq[Expression], Projection] {
         public void setNullAt(int i) { nullBits[i] = true; }
         public boolean isNullAt(int i) { return nullBits[i]; }
 
-        protected Object genericGet(int i) {
+        @Override
+        public Object genericGet(int i) {
```

Comment:
I made `SpecificRow` extends `CodeGenMutableRow` which mix-in `BaseGenericInternalRow`, so that we don't need to codegen all special getters inside `SpecificRow`.
The `genericGetter` should be `protected` here, but seems it's a scala mix-in bug that I have to use public.